### PR TITLE
script: dont unwrap in header set

### LINKED ISSUE #32968: Panic in fetch/api/headers/headers-normalize.any.js.ini
To reproduce:
* `./mach test-wpt tests/wpt/tests/fetch/api/headers/headers-normalize.any.js`

Backtrace:

<details>

```
pid:9842 called `Result::unwrap()` on an `Err` value: InvalidHeaderValue (thread Script(1,1), at components/script/dom/headers.rs:219)
 0:06.82 pid:9842    0: backtrace::backtrace::libunwind::trace
 0:06.82 pid:9842              at /Users/jdm/.cargo/registry/src/index.crates.io-6f17d22bba15001f/backtrace-0.3.73/src/backtrace/libunwind.rs:116:5
 0:06.82 pid:9842       backtrace::backtrace::trace_unsynchronized
 0:06.82 pid:9842              at /Users/jdm/.cargo/registry/src/index.crates.io-6f17d22bba15001f/backtrace-0.3.73/src/backtrace/mod.rs:66:5
 0:06.82 pid:9842    1: <servoshell::backtrace::Print as core::fmt::Debug>::fmt
 0:06.82 pid:9842              at /Users/jdm/src/servo/ports/servoshell/backtrace.rs:65:13
 0:06.83 pid:9842    2: core::fmt::rt::Argument::fmt
 0:06.83 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/fmt/rt.rs:142:9
 0:06.83 pid:9842       core::fmt::write
 0:06.83 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/fmt/mod.rs:1153:17
 0:06.84 pid:9842    3: std::io::Write::write_fmt
 0:06.84 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/io/mod.rs:1843:15
 0:06.84 pid:9842    4: servoshell::backtrace::print
 0:06.84 pid:9842              at /Users/jdm/src/servo/ports/servoshell/backtrace.rs:18:5
 0:06.84 pid:9842    5: servoshell::panic_hook::panic_hook
 0:06.84 pid:9842              at /Users/jdm/src/servo/ports/servoshell/panic_hook.rs:38:17
 0:06.86 pid:9842    6: core::ops::function::Fn::call
 0:06.86 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/ops/function.rs:79:5
 0:06.91 pid:9842    7: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call
 0:06.91 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/alloc/src/boxed.rs:2034:9
 0:06.91 pid:9842       std::panicking::rust_panic_with_hook
 0:06.91 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:783:13
 0:06.91 pid:9842    8: std::panicking::begin_panic_handler::{{closure}}
 0:06.91 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:657:13
 0:06.91 pid:9842    9: std::sys_common::backtrace::__rust_end_short_backtrace
 0:06.91 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:171:18
 0:06.91 pid:9842   10: rust_begin_unwind
 0:06.91 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:645:5
 0:06.91 pid:9842   11: core::panicking::panic_fmt
 0:06.91 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/panicking.rs:72:14
 0:06.92 pid:9842   12: core::result::unwrap_failed
 0:06.92 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/result.rs:1654:5
 0:06.98 pid:9842   13: core::result::Result<T,E>::unwrap
 0:06.98 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/result.rs:1077:23
 0:06.98 pid:9842       <script::dom::headers::Headers as script::dom::bindings::codegen::Bindings::HeadersBinding::Headers_Binding::HeadersMethods>::Set
 0:06.98 pid:9842              at /Users/jdm/src/servo/components/script/dom/headers.rs:219:13
 0:07.01 pid:9842   14: script::dom::bindings::codegen::Bindings::HeadersBinding::Headers_Binding::set::{{closure}}::{{closure}}
 0:07.01 pid:9842              at /Users/jdm/src/servo/target/debug/build/script-84e8aed3a23eae36/out/Bindings/HeadersBinding.rs:353:41
 0:07.01 pid:9842   15: script::dom::bindings::codegen::Bindings::HeadersBinding::Headers_Binding::set::{{closure}}
 0:07.01 pid:9842              at /Users/jdm/src/servo/target/debug/build/script-84e8aed3a23eae36/out/Bindings/HeadersBinding.rs:323:33
 0:07.01 pid:9842   16: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &mut F>::call_once
 0:07.01 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/ops/function.rs:305:13
 0:07.02 pid:9842   17: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
 0:07.02 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/panic/unwind_safe.rs:272:9
 0:07.02 pid:9842   18: std::panicking::try::do_call
 0:07.02 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:552:40
 0:07.02 pid:9842   19: ___rust_try
 0:07.02 pid:9842   20: std::panicking::try
 0:07.02 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:516:19
 0:07.03 pid:9842   21: std::panic::catch_unwind
 0:07.03 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panic.rs:146:14
 0:07.03 pid:9842   22: mozjs::panic::wrap_panic
 0:07.03 pid:9842              at /Users/jdm/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/fb8225e/mozjs/src/panic.rs:22:11
 0:07.05 pid:9842   23: script::dom::bindings::codegen::Bindings::HeadersBinding::Headers_Binding::set
 0:07.05 pid:9842              at /Users/jdm/src/servo/target/debug/build/script-84e8aed3a23eae36/out/Bindings/HeadersBinding.rs:323:5
 0:07.09 pid:9842   24: CallJitMethodOp
 0:07.09 pid:9842              at /Users/runner/work/mozjs/mozjs/mozjs-sys/src/jsglue.cpp:616:10
 0:07.12 pid:9842   25: script::dom::bindings::utils::generic_call
 0:07.12 pid:9842              at /Users/jdm/src/servo/components/script/dom/bindings/utils.rs:520:5
 0:07.12 pid:9842   26: script::dom::bindings::utils::generic_method
 0:07.12 pid:9842              at /Users/jdm/src/servo/components/script/dom/bindings/utils.rs:536:5
 0:07.12 pid:9842   27: __ZN2js23InternalCallOrConstructEP9JSContextRKN2JS8CallArgsENS_14MaybeConstructENS_10CallReasonE
 0:07.12 pid:9842   28: __ZN2js9InterpretEP9JSContextRNS_8RunStateE
 0:07.12 pid:9842   29: __ZN2js9RunScriptEP9JSContextRNS_8RunStateE
 0:07.12 pid:9842   30: __ZN2js23InternalCallOrConstructEP9JSContextRKN2JS8CallArgsENS_14MaybeConstructENS_10CallReasonE
 0:07.12 pid:9842   31: __ZN2js4CallEP9JSContextN2JS6HandleINS2_5ValueEEES5_RKNS_13AnyInvokeArgsENS2_13MutableHandleIS4_EENS_10CallReasonE
 0:07.12 pid:9842   32: __ZN2js9fun_applyEP9JSContextjPN2JS5ValueE
 0:07.12 pid:9842   33: __ZN2js23InternalCallOrConstructEP9JSContextRKN2JS8CallArgsENS_14MaybeConstructENS_10CallReasonE
 0:07.12 pid:9842   34: __ZN2js9InterpretEP9JSContextRNS_8RunStateE
 0:07.12 pid:9842   35: __ZN2js9RunScriptEP9JSContextRNS_8RunStateE
 0:07.12 pid:9842   36: __ZN2js7ExecuteEP9JSContextN2JS6HandleIP8JSScriptEENS3_IP8JSObjectEENS2_13MutableHandleINS2_5ValueEEE
 0:07.12 pid:9842   37: __Z16JS_ExecuteScriptP9JSContextN2JS6HandleIP8JSScriptEENS1_13MutableHandleINS1_5ValueEEE
 0:07.15 pid:9842   38: mozjs::rust::wrappers::JS_ExecuteScript
 0:07.15 pid:9842              at /Users/jdm/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/fb8225e/mozjs/src/rust.rs:1080:51
 0:07.18 pid:9842   39: script::dom::globalscope::GlobalScope::evaluate_script_on_global_with_result::{{closure}}
 0:07.18 pid:9842              at /Users/jdm/src/servo/components/script/dom/globalscope.rs:2716:34
 0:07.21 pid:9842   40: profile_traits::time::profile
 0:07.21 pid:9842              at /Users/jdm/src/servo/components/shared/profile/time.rs:147:15
 0:07.24 pid:9842   41: script::dom::globalscope::GlobalScope::evaluate_script_on_global_with_result
 0:07.24 pid:9842              at /Users/jdm/src/servo/components/script/dom/globalscope.rs:2645:9
 0:07.27 pid:9842   42: script::dom::htmlscriptelement::HTMLScriptElement::run_a_classic_script
 0:07.27 pid:9842              at /Users/jdm/src/servo/components/script/dom/htmlscriptelement.rs:1067:9
 0:07.27 pid:9842   43: script::dom::htmlscriptelement::HTMLScriptElement::execute
 0:07.27 pid:9842              at /Users/jdm/src/servo/components/script/dom/htmlscriptelement.rs:1029:17
 0:07.30 pid:9842   44: script::dom::servoparser::ServoParser::resume_with_pending_parsing_blocking_script
 0:07.30 pid:9842              at /Users/jdm/src/servo/components/script/dom/servoparser/mod.rs:327:9
 0:07.34 pid:9842   45: script::dom::document::Document::process_pending_parsing_blocking_script
 0:07.34 pid:9842              at /Users/jdm/src/servo/components/script/dom/document.rs:2509:13
 0:07.34 pid:9842   46: script::dom::document::Document::pending_parsing_blocking_script_loaded
 0:07.34 pid:9842              at /Users/jdm/src/servo/components/script/dom/document.rs:2495:9
 0:07.34 pid:9842   47: script::dom::htmlscriptelement::finish_fetching_a_classic_script
 0:07.34 pid:9842              at /Users/jdm/src/servo/components/script/dom/htmlscriptelement.rs:329:13
 0:07.34 pid:9842   48: <script::dom::htmlscriptelement::ClassicContext as net_traits::FetchResponseListener>::process_response_eof
 0:07.34 pid:9842              at /Users/jdm/src/servo/components/script/dom/htmlscriptelement.rs:470:13
 0:07.37 pid:9842   49: <net_traits::FetchResponseMsg as net_traits::Action<T>>::process
 0:07.37 pid:9842              at /Users/jdm/src/servo/components/shared/net/lib.rs:304:25
 0:07.40 pid:9842   50: <script::network_listener::ListenerTask<A,Listener> as script::task::TaskOnce>::run_once
 0:07.40 pid:9842              at /Users/jdm/src/servo/components/script/network_listener.rs:117:13
 0:07.43 pid:9842   51: <script::task::CancellableTask<T> as script::task::TaskOnce>::run_once
 0:07.43 pid:9842              at /Users/jdm/src/servo/components/script/task.rs:114:13
 0:07.43 pid:9842   52: <T as script::task::TaskBox>::run_box
 0:07.43 pid:9842              at /Users/jdm/src/servo/components/script/task.rs:57:9
 0:07.46 pid:9842   53: script::script_thread::ScriptThread::handle_msg_from_script
 0:07.46 pid:9842              at /Users/jdm/src/servo/components/script/script_thread.rs:2479:17
 0:07.50 pid:9842   54: script::script_thread::ScriptThread::handle_msgs::{{closure}}
 0:07.50 pid:9842              at /Users/jdm/src/servo/components/script/script_thread.rs:1958:46
 0:07.50 pid:9842   55: script::script_thread::ScriptThread::profile_event
 0:07.50 pid:9842              at /Users/jdm/src/servo/components/script/script_thread.rs:2214:13
 0:07.50 pid:9842   56: script::script_thread::ScriptThread::handle_msgs
 0:07.50 pid:9842              at /Users/jdm/src/servo/components/script/script_thread.rs:1951:26
 0:07.50 pid:9842   57: script::script_thread::ScriptThread::start
 0:07.50 pid:9842              at /Users/jdm/src/servo/components/script/script_thread.rs:1454:15
 0:07.50 pid:9842   58: <script::script_thread::ScriptThread as script_layout_interface::ScriptThreadFactory>::create::{{closure}}::{{closure}}
 0:07.50 pid:9842              at /Users/jdm/src/servo/components/script/script_thread.rs:845:25
 0:07.53 pid:9842   59: profile_traits::mem::ProfilerChan::run_with_memory_reporting
 0:07.53 pid:9842              at /Users/jdm/src/servo/components/shared/profile/mem.rs:91:9
 0:07.53 pid:9842   60: <script::script_thread::ScriptThread as script_layout_interface::ScriptThreadFactory>::create::{{closure}}
 0:07.53 pid:9842              at /Users/jdm/src/servo/components/script/script_thread.rs:843:17
 0:07.56 pid:9842   61: std::sys_common::backtrace::__rust_begin_short_backtrace
 0:07.56 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:155:18
 0:07.59 pid:9842   62: std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}
 0:07.59 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/thread/mod.rs:528:17
 0:07.62 pid:9842   63: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
 0:07.62 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/panic/unwind_safe.rs:272:9
 0:07.65 pid:9842   64: std::panicking::try::do_call
 0:07.65 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:552:40
 0:07.65 pid:9842   65: ___rust_try
 0:07.65 pid:9842   66: std::panicking::try
 0:07.65 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:516:19
 0:07.65 pid:9842   67: std::panic::catch_unwind
 0:07.65 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panic.rs:146:14
 0:07.65 pid:9842       std::thread::Builder::spawn_unchecked_::{{closure}}
 0:07.65 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/thread/mod.rs:527:30
 0:08.04 pid:9842   68: core::ops::function::FnOnce::call_once{{vtable.shim}}
 0:08.04 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/ops/function.rs:250:5
 0:08.04 pid:9842   69: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
 0:08.04 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/alloc/src/boxed.rs:2020:9
 0:08.04 pid:9842       <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
 0:08.04 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/alloc/src/boxed.rs:2020:9
 0:08.04 pid:9842       std::sys::pal::unix::thread::Thread::new::thread_start
 0:08.04 pid:9842              at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys/pal/unix/thread.rs:108:17
 0:08.05 pid:9842   70: __pthread_joiner_wake
```

</details>

This points to https://github.com/servo/servo/blob/main/components/script/dom/headers.rs#L219 . We should not be unwrapping a n observable fallible operation like this. Let's follow the example from the Append method in the same file, which explicitly handles the Err case by silently skipping the operation.

### PR DESCRIPTION
**Pull request by ToBinio**
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #32968 (GitHub issue number if applicable)

- [X] There are tests for these changes OR
- [ ] These changes do not require tests because ___

please note that this is my first PR here so i might have done something wrong. I simply saw a small issue I thought I try to fix.

### CODE REVIEW

**Comment by jdm** [comment]
Ah, looks like those are the tests that were triggering the panic before. We should investigate what it takes to get them passing now!

File: components/script/dom/headers.rs [hunk 1]
```diff
@@ -214,10 +214,20 @@ impl HeadersMethods for Headers {
...
-        );
+        match (
+            HeaderName::from_str(&valid_name),
+            HeaderValue::from_bytes(&valid_value),
+        ) {
+            (Ok(key), Ok(val)) => {
+                self.header_list.borrow_mut().insert(key, val);
+            },
+            _ => {
+                warn!(
```
Comment (Taym95) [inline components/script/dom/headers.rs:225]: should we return error in this case?

**Comment by ToBinio** [comment]
> Ah, looks like those are the tests that were triggering the panic before. We should investigate what it takes to get them passing now!

the "Stable unexpected results (2)" are caused by [normalize_value](https://github.com/servo/servo/blob/08eb4faf4d2805283137a19739b092cd7ddff600/components/script/dom/headers.rs#L412) not handling this case correctly:

 ` "name7": ["\t\f\tnewLine\n", "\f\tnewLine"],`
 
 should this be fixed in this PR?
 
 What about the other failing tests? they often don't fail on my system, do I need to change something on my system or are they just flaky?

**Comment by jdm** [comment]
We only care about the stable unexpected results that you pointed out. https://wpt.fyi/results/fetch/api/headers/headers-normalize.any.html?label=master&label=experimental&aligned&q=header%20normalize says that other browsers are passing it, so it would be good to fix our behavior in this PE, if you're up for that!

**Comment by ToBinio** [comment]
The problem is that http HeaderValue doesn't allow `\f\t` to be in it. it only allowed visible ASCII characters. But the test wants them.

**Comment by jdm** [comment]
Ok, sounds like we shouldn't hold up this PR for that, since we'll need to convince the owners of the http crate to change the behaviour. We can mark them as expected failures in the test's ini file.

**Review by jdm — APPROVED** [review]
Thank you!

**Comment by ToBinio** [comment]
done,
Btw, Thanks for the super quick & helpful answers

**Pull request merged**